woocommerce/woocommerce-gateway-stripe
# [Klarna] Only display it when the presentment currency matches the currency of the customer’s country

Fixes #3118 

For transactions between the EEA, UK, and Switzerland, the presentment currency [must match the currency of the customer’s country](https://docs.stripe.com/payments/klarna#:~:text=Swedish%20business%20can%20present%20in%20EUR%20to%20accept%20Klarna%20from%20a%20buyer%20in%20Germany).

Currently, we display this payment method on the checkout page for all these billing countries, but the presentment currency matches the Stripe account's country, not the customer's.

When this happens, the following error is displayed on the checkout page:

`There was an error processing the payment: The provided PaymentMethod cannot be attached. To reuse a PaymentMethod, you must attach it to a Customer first.`

## Changes proposed in this Pull Request:
- In `get_supported_currencies` function, returning the currency based on the customer billing address for EEA, UK, and Switzerland.

## Testing instructions

**Steps to reproduce**

1. Checkout to `develop` branch.
2. Connect a Stripe account from the UK.
3. Select Pound sterling as the store currency.
4. Enable Klarna.
5. As a shopper, add a product to the page and proceed to checkout.
6. Select France as the billing country.
7. Notice Klarna continues to be displayed as an available payment method.
8. Place the order using Klarna.
9. Notice the error message mentioned about is displayed.

**Steps to test this PR**
1. Checkout to `fix/3118-klarna-currency` branch.
2. Connect a Stripe account from the UK.
3. Select Pound sterling as the store currency.
4. Enable Klarna.
5. As a shopper, add a product to the page and proceed to checkout.
6. Select France as the billing country.
7. Notice that Klarna is not available as a payment method now because for France the currency is Euro which is not same as the presentment currency Pound.
8. Now connect a Stripe account from any EEA country.
9. Select Euro as the store currency.
10. As a shopper, add a product to the page and proceed to checkout.
11. Select France as the billing country.
12. Notice that Klarna is present as an available payment method.
13. Select Germany as the billing country.
14. Notice that Klarna is present as an available payment method.
15. Select UK as the billing country. Notice that Klarna is not available as a payment method now because for UK the currency is Pound which is not same as the presentment currency Euro.

**Note**
I will fix the tests
